src/components/CustomSelect.js

```diff
@@ -17,22 +17,24 @@ function CustomSelect({ optionsList, selectedOption, onChange }) {
         className={`toggle ${isOptionsOpen ? 'expanded' : ''}`}
         onClick={toggleOptions}
       >
-        {selectedOption}
+        {selectedOption.name}
```

Comment:
`function CustomSelect({ optionsList, selectedOption, onChange }) {`
please find a better name for onChange. onChange of what? How onChange can be in onClick?